### PR TITLE
US004 - inserted new sales employee

### PR DESCRIPTION
**Pull request by ManpreetM01**

### CODE REVIEW

File: src/main/java/org/kainos/ea/api/SalesService.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import org.kainos.ea.cli.SalesEmployeeRequest;
+import org.kainos.ea.client.FailedToCreateSalesEmployeeException;
+import org.kainos.ea.cli.SalesEmployee;
+import org.kainos.ea.client.InvalidSalesEmployeeException;
+import org.kainos.ea.db.SalesDAO;
+
+import java.sql.SQLException;
+
+public class SalesService {
+    private SalesDAO salesDao = new SalesDAO();
```
Comment (shaunganley) [inline src/main/java/org/kainos/ea/api/SalesService.java]: pass this in via a constructor

File: src/main/java/org/kainos/ea/api/SalesService.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    public int createSalesEmployee(SalesEmployeeRequest salesEmployee) throws FailedToCreateSalesEmployeeException {
+        try {
+            int id = salesDao.createSalesEmployee(salesEmployee);
```
Comment (shaunganley) [inline src/main/java/org/kainos/ea/api/SalesService.java:33]: should we validate the request object before storing in the db?

File: src/main/java/org/kainos/ea/cli/Employee.java [hunk 3]
```diff
@@ -1,6 +1,9 @@
 package org.kainos.ea.cli;
 
 public class Employee {
+    public Employee(int id, String name, double salary, String bankAccountNumber, String nationInsuranceNumber) {
```
Comment (shaunganley) [inline src/main/java/org/kainos/ea/cli/Employee.java:5]: you need to set the values in here

File: src/main/java/org/kainos/ea/db/SalesDAO.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+import java.sql.*;
+
+public class SalesDAO {
+    private DbConnector databaseConnector = new DbConnector();
+
+    public int createSalesEmployee(SalesEmployeeRequest salesEmployee) throws SQLException {
+        Connection c = databaseConnector.getConnection();
+
+        String insertStatement = "INSERT INTO Sales_Employee (Name, Salary, bank_number, ni, commission_rate) VALUES (?,?,?,?,?)";
```
Comment (shaunganley) [inline src/main/java/org/kainos/ea/db/SalesDAO.java:16]: do you not have an employee table that also needs updated?

File: src/main/java/org/kainos/ea/db/SalesDAO.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+            throw new ProductDoesNotExistException();
+        }
+
+        productDao.updateProduct(id, product);
+
+    } catch (SQLException e) {
+        System.err.println(e.getMessage());
+        throw new FailedToUpdateProductException();
+    }
+    }*/
```
Comment (shaunganley) [inline src/main/java/org/kainos/ea/db/SalesDAO.java]: remove if not used

File: src/main/java/org/kainos/ea/resources/SalesEmployeeController.java [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+@Path("/api")
+public class SalesEmployeeController {
+    private SalesService salesService = new SalesService();
+
+    @POST
+    @Path("/sales_employees")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response createSalesEmployee(SalesEmployeeRequest salesEmployee) {
+        try {
+            return Response.ok(salesService.createSalesEmployee(salesEmployee)).build();
```
Comment (shaunganley) [inline src/main/java/org/kainos/ea/resources/SalesEmployeeController.java]: should return a 201 rather than 200 response code

**Review by shaunganley — CHANGES_REQUESTED** [review]

File: src/main/java/org/kainos/ea/db/SalesDAO.java (hunk 4 above)
Comment (ManpreetM01) [reply src/main/java/org/kainos/ea/db/SalesDAO.java:16]: No

File: src/main/java/org/kainos/ea/resources/SalesEmployeeController.java (hunk 6 above)
Comment (ManpreetM01) [reply src/main/java/org/kainos/ea/resources/SalesEmployeeController.java]: made changes

File: src/main/java/org/kainos/ea/cli/Employee.java (hunk 3 above)
Comment (ManpreetM01) [reply src/main/java/org/kainos/ea/cli/Employee.java:5]: chagnes made

File: src/main/java/org/kainos/ea/api/SalesService.java (hunk 2 above)
Comment (ManpreetM01) [reply src/main/java/org/kainos/ea/api/SalesService.java:33]: channges made

File: src/main/java/org/kainos/ea/api/SalesService.java (hunk 1 above)
Comment (ManpreetM01) [reply src/main/java/org/kainos/ea/api/SalesService.java]: changes made

File: src/main/java/org/kainos/ea/db/SalesDAO.java (hunk 5 above)
Comment (ManpreetM01) [reply src/main/java/org/kainos/ea/db/SalesDAO.java]: changes made

**Review by TomBillingtonUK — APPROVED** [review]

**Pull request merged**